backends/bmv2/control.cpp

```diff
@@ -342,32 +326,18 @@ ControlConverter::convertTable(const CFG::TableNode* node,
                 int l = slice->getL();
                 mask = Util::maskFromSlice(h, l);
             }
-            if (match_type == "valid") {
-                auto mt = refMap->getDeclaration(ke->matchType->path, true)
-                          ->to<IR::Declaration_ID>();
-                if (expr->is<IR::MethodCallExpression>()) {
-                    auto mi = P4::MethodInstance::resolve(expr->to<IR::MethodCallExpression>(),
-                                            refMap, typeMap);
-                    if (mi->is<P4::BuiltInMethod>()) {
-                        auto bim = mi->to<P4::BuiltInMethod>();
-                        if (bim->name == IR::Type_Header::isValid) {
-                            if (mt->name.name == backend->getCoreLibrary().ternaryMatch.name) {
-                                expr = new IR::Member(IR::Type::Boolean::get(), bim->appliedTo,
-                                                      "$valid$");
-                                typeMap->setType(expr, expr->type);
-                                match_type = backend->getCoreLibrary().ternaryMatch.name;
-                                if (match_type != table_match_type &&
-                                    table_match_type !=
-                                        BMV2::MatchImplementation::rangeMatchTypeName)
-                                    table_match_type =
-                                        backend->getCoreLibrary().ternaryMatch.name;
-                            } else {
-                                expr = bim->appliedTo; }}}}
-            }
 
             auto keyelement = new Util::JsonObject();
             keyelement->emplace("match_type", match_type);
-            auto jk = conv->convert(expr);
+
+            // XXX(seth): Boolean variables are stored as data, so we normally
```

Comment:
Because the name "convertLeftValue()" doesn't really capture the reason we're calling the function. It needs a better name, or perhaps we need a different function with a clearer name, but I couldn't immediately think of one.